apps/dashboard/app/apps/user_configuration.rb

```diff
@@ -0,0 +1,36 @@
+class UserConfiguration
+
+  def initialize(config)
+    @config = config ? config.deep_symbolize_keys : {}
+  end
+
+  # The dashboard's landing page layout. Defaults to nil.
+  def dashboard_layout
+    fetch(:dashboard_layout, nil)
+  end
+
+  # The configured pinned apps
+  def pinned_apps
+    fetch(:pinned_apps, [])
+  end
+
+  # The length of the "Pinned Apps" navbar menu
+  def pinned_apps_menu_length
+    fetch(:pinned_apps_menu_length, 6)
+  end
+
+  def profile_links
+    fetch(:profile_links, [])
+  end
```

Comment:
Instead of a new configuration option for links, can we embed the information into the `profile` struct directly? Navbar is obviously in flux with all of this, but in any case - I'd like to cut down on configuration complexity for folks writing these configs.

```yaml
profiles:
  first:
    icon: cog
  next:
     icon: beaker
```

```ruby
def profile_links
   profiles.map do |profile|
     { icon: profile[:icon] ... }
   end
end
```
I'm not really sure what you're doing with navigation, but I'd like to make the help menu a lot more dynamic anyhow in #1909, so that bit's due to change anyhow.

Reply:
For the profile links, this was done on purpose. We wanted decouple how the profile links are displayed from the definition. It is a bit of an overhead to create the array, but will allow to create a different list per profile if needed.

As for the other features like navigation items, look and feel, ... we haven't thought of an implementation just yet. We thought that these will be added individually later with their own issues. To make small incremental changes that should be easier to review and merge.

For this change, we planned on just update the pinned_apps and the dashboard_layout.

Reply:
> but will allow to create a different list per profile if needed

OK, we'll defer it but my guess is when we start building navbar groups it'll kinda be easy to merge them.